transport/src/main/java/io/netty/channel/DefaultChannelId.java

```diff
@@ -189,48 +189,59 @@ static int defaultProcessId() {
     private transient String longValue;
 
     private DefaultChannelId() {
-        data = new byte[MACHINE_ID.length + PROCESS_ID_LEN + SEQUENCE_LEN + TIMESTAMP_LEN + RANDOM_LEN];
+        final byte[] data = new byte[MACHINE_ID.length + PROCESS_ID_LEN + SEQUENCE_LEN + TIMESTAMP_LEN + RANDOM_LEN];
         int i = 0;
 
         // machineId
         System.arraycopy(MACHINE_ID, 0, data, i, MACHINE_ID.length);
         i += MACHINE_ID.length;
 
         // processId
-        i = writeInt(i, PROCESS_ID);
+        writeInt(data, i, PROCESS_ID);
+        i += Integer.BYTES;
 
         // sequence
-        i = writeInt(i, nextSequence.getAndIncrement());
+        writeInt(data, i, nextSequence.getAndIncrement());
+        i += Integer.BYTES;
 
         // timestamp (kind of)
-        i = writeLong(i, Long.reverse(System.nanoTime()) ^ System.currentTimeMillis());
+        writeLong(data, i, Long.reverse(System.nanoTime()) ^ System.currentTimeMillis());
+        i += Long.BYTES;
 
         // random
         int random = PlatformDependent.threadLocalRandom().nextInt();
-        i = writeInt(i, random);
+        writeInt(data, i, random);
+        i += Integer.BYTES;
         assert i == data.length;
 
+        this.data = data;
         hashCode = Arrays.hashCode(data);
     }
 
-    private int writeInt(int i, int value) {
-        data[i ++] = (byte) (value >>> 24);
-        data[i ++] = (byte) (value >>> 16);
-        data[i ++] = (byte) (value >>> 8);
-        data[i ++] = (byte) value;
-        return i;
+    private static void writeInt(byte[] data, int i, int value) {
+        if (PlatformDependent.isUnaligned()) {
+            PlatformDependent.putInt(data, i, value);
```

Comment:
how about the endianess? I don't remember by heart if putInt is BE, if it is, it's ok

Reply:
oh, let me fix it.

Reply:
I changed it.